R/eatModel.r

```diff
@@ -1,94 +1,59 @@
-    x <- unlist ( strsplit(x, " +") )
-    return(x)  } )
-  return(isL)}
+  isL <- lapply(splt, FUN = function(x) {
+    ind <- which(x %in% c(letters, LETTERS))
+    x[setdiff(1:length(x), ind)] <- " "
+    x <- eatTools::crop(paste(x, sep = "", collapse = ""))
+    x <- unlist(strsplit(x, " +"))
+    return(x)
+  })
+  return(isL)
+}
 
 ### not called ----------------------------------------------------------------------------
 
-simEquiTable <- function ( anchor, mRef, sdRef, addConst = 500, multConst = 100, cutScores) {
-                anchor<- eatTools::makeDataFrame(anchor)
-                if ( ncol(anchor) != 2) {
-                     warning(paste0("'anchor' has ",ncol(anchor)," columns. First column is used as item ID, second column is used as item parameter."))
-                }
-                if(!inherits(anchor[,2], c("integer", "numeric"))) {stop("Item parameter column must be numeric.")}
-                if(length(unique(anchor[,1])) != nrow(anchor)) {stop("Item ID column has duplicated entries.")}
-                dtmp  <- data.frame(rbind(1*(lower.tri(matrix(1, nrow = nrow(anchor), ncol = nrow(anchor)))),1))
-                dtmp  <- data.frame(dtmp, score = rowSums(dtmp) , irtoys::wle(dtmp, cbind(1, anchor[,2], 0)), stringsAsFactors = FALSE)
-                dtmp[,"bista"] <- (dtmp[,"est"] - mRef) / sdRef * multConst + addConst
-                dtmp[,"ks"]    <- eatTools::num.to.cat ( x = dtmp[,"bista"], cut.points = cutScores[["values"]], cat.values = cutScores[["labels"]])
-                dig   <- 0
-                while ( length(which(round(dtmp[,"bista"], digits = dig) %in% cutScores[["values"]])) > 0) {dig <- dig + 1}
-                shrt  <- do.call("rbind", by ( data = dtmp, INDICES = dtmp[,"ks"], FUN = function ( sks ) { data.frame ( score = paste(c(min(sks[,"score"]), max(sks[,"score"])), collapse=" bis "), estimate = paste(round(c(min(sks[,"est"]), max(sks[,"est"])),digits=2), collapse=" bis "), bista = paste(round(c(min(sks[,"bista"]), max(sks[,"bista"])),digits=dig), collapse=" bis "), ks=unique(sks[,"ks"]), stringsAsFactors=FALSE)}))
-                return(list ( complete = dtmp[,c("score", "est", "bista", "ks")], short = shrt))}
+simEquiTable <- function(anchor, mRef, sdRef, addConst = 500, mulConst = 100, cutScores){
+  # checks/prep
+  anchor <- eatTools::makeDataFrame(anchor)
+  if(ncol(anchor) != 2){
+    warning(paste0("'anchor' has ", ncol(anchor)," columns. First column is used as item ID, second column is used as item parameter."))
+  }
+  if(!inherits(anchor[,2], c("integer", "numeric"))){
+    stop("Item parameter column must be numeric.")
+  }
+  if(length(unique(anchor[,1])) != nrow(anchor)){
+    stop("Item ID column has duplicated entries.")
+  }
+  checkmate::assert_numeric(mRef, len = 1)
+  lapply(c(sdRef, addConst, mulConst), checkmate::assert_numeric, len = 1, lower = 0)
+  checkmate::assert_list(cutScores, min.len = 1, max.len = 2)

```

Comment:
When cutScores only has the values argument without labels, does it still have to be a list?

Reply:
Yes, it must be a list in any case. The reason is that - if internal functions retrieve the concrete cut scores - the call `cutScores[["values"]]` should always be used. Otherwise you would have to check each time: is `cutScores` a list? Then use `cutScores[["values"]]`, otherwise use `cutScores`. That would be somehow inconsistent.